lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -457,6 +478,19 @@ def _container_security_context_is_set(
         return True
 
 
+class NadConfigChangedEvent(EventBase):
+    """Event triggered when an existing network attachment definition is changed."""
+
+    def __init__(self, handle: Handle):
+        super().__init__(handle)
+
+
+class NadConfigChangedCharmEvents(CharmEvents):
+    """NAD config changed events."""
+
+    nad_config_changed = EventSource(NadConfigChangedEvent)
+
+
```

Comment:
I'm not sure we need to have this at all, users should be able to pass any event (ex. ConfigChanged or a custom event) and our code should not care.

Reply:
There is a reason to define the a custom events in the library part. The event to be observed will be a custom event as native charm events does not appropriate for our situation. If we do not define any custom event in the library side, each consumer charm needs to define custom events to be passed and observed. This requires more modifications and repeated work in the several consumer charms. 
If a custom event exist in the library part which is created for this purpose, charms can import and use it easily.

Reply:
This is removed from the library side.